common/downstream_fatal_error.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: default
+parent: Common
+---
+
+# Downstream Fatal Error
+
+Understanding the `DownstreamFatalError` exception.
+{: .fs-6 .fw-300 }
+
+&nbsp;
+
+{: .note }
+> _pyRACF expects IRRSMO00 to return a string containing XML data. If IRRSMO00 returns an empty string, or if the **SAF Return Code** in the result XML is greater than `4`, `DownstreamFatalError` is raised._
+
+&nbsp;
+
+pyRACF expects IRRSMO00 to return a non-empty response string after processing a request. If IRRSMO00 returns an empty response string or the **SAF Return Code** is greater than `4`, indicating that there was an issue with command image processing, a `DownstramFatalError` will be raised. It is possible that RACF commands are executed in this situation, but this is not the case for any known causes of this error.
+
+## RACF Authorizations
+
+&nbsp;
+
+A common cause of this error is that the user does not have the proper RACF authorizations as outlined in [our dependencies note](../../index). One of the possible failures is the user not having `READ` authority to the `IRR.IRRSMO00.PRECHECK` resource in the `XFACILIT` class, which is required for `set` or `alter` operations. If you are not certain if you have this authority or if this has been established in your environment, please consult our [Check for & set up RACF Authorizations](../setup_precheck) documentation. Another possibility is the user not having `ALTER` authority to the `userid.IRRSMO00` resource in the `SURROGAT` class, which is required to run pyRACF commands as another userid. Please review our [run as userid](../run_as_userid) documentation for more information on this feature.

```

Comment:
```markdown
at least `READ` authority.
```

Reply:
Change `run pyRACF commands` to `make security requests`.

Reply:
changes made.

Reply:
Change
```markdown
If you are not certain if you have this authority or if this has been established in your environment, please consult our [Check for & set up RACF Authorizations](../setup_precheck) documentation.
```
to
```markdown
See the [Check for & set up RACF Authorizations](../setup_precheck) documentation for more details.
```

Reply:
change made.

Reply:
Change `One of the possible failures is` to `One of the possible causes for this error`.

Change `Another possibility is the` to `Another possible cause of this error is the`

Reply:
good point. change made.

Reply:
Change `caused` to `causes`.